### DISCUSSION TITLE
Ping/Pong Mismatch on Interface

### DISCUSSION THREAD

**Discussion by SL552114** [SukramJ/aiohomematic#1292] [General]
Hallo alle!

Ich versuche nun leider seit drei Monaten erfolglos dieses Problem zu beheben.
Ich habe meine CCU schon zahlreiche Male neu gestartet - ich habe gedacht, dass es sich damit sogar erledigt hatte. Aber es kommt leider immer wieder.
Nun suche ich schon eine lange Weile nach einer anderen Lösung. Im Hilfetext ist beschrieben, dass man schauen soll, ob man zwei HA Instanzen mit dem gleichen Namen betreibt. Ich nehme an, es ist gemeint, dass ich eine zweite HM Integration mit dem gleichen Namen betreibe. Aber ich habe keine Ahnung, wie ich das herausfinden kann.
Ich hoffe wirklich, dass es eine Lösung gibt, ohne, dass ich alles nochmal neu einrichte.
Könnte mir jemand dazu eine Hilfestellung geben? Wo sehe ich diese zweite Instanz? Unter den deaktivierten Integrationen sehe ich schonmal nichts.

Vielen Dank für die wirklich super Arbeit an dieser Integration und natürlich auch für eine Hilfestellung zu diesem Problem!

Holger

**Comment by SukramJ** [comment]
Wie lange nach einem HA  Neustart dauert es denn, bis die Meldung erscheint?

>  Ich nehme an, es ist gemeint, dass ich eine zweite HM Integration mit dem gleichen Namen betreibe

Ich gehe davon aus, das Du weißt, ob Du eine weitere HA Installation irgendwo betreibst.
Hier geht es darum, ob eine zweite (oder mehr) HA Installation existiert, die diese Integration mit dem gleichen Instanznamen verwendet.  In dem Fall bekommt die zuletzt gestartete HA installation alle Events, und die erste bekommt keine. 

Es gibt aber auch den Fall, das eine CCU, warum auch immer,  Events "verschluckt". Für Ping/Pong werden syntetische Events (PING) an die CCU geschickt, auf die die CCU **immer** mit einem PONG antwortet. Dadurch kann geprüft werden ob die HA/CCU Kommunikation funktioniert.  Wenn es hierbei zu Problemen kommt ist auch davon auszugehen, das die eigentlichen Geräteaktualisierung nicht, oder nicht alle, an HA gesendet werden konnten, und das wäre dann das eigentliche Problem.
Hier sollte man mal schauen ob es irgendwelche Auffälligkeiten in den CCU Logs gibt.

Wenn Du aber keine Probleme mit ausbleibenden Geräteaktualisierungen in HA feststellen konntest, dann bleibt dir als letzter Ausweg noch die Möglichkeit die Benachrichtigungen in der Konfiguration zu deaktivieren.

**Reply by SL552114** [reply]
Hi @SukramJ, danke für deine schnelle Antwort.

Gestern habe ich gerade auf 2023.11.2 upgedated. Seit dem ist es bisher zum Beispiel nicht wieder aufgetreten. Du liegst mit deiner Frage genau richtig: Wenn es weg ist, dann dauert es immer ein wenig, bis es wiederkommt.
Deswegen habe ich in der Vergangenheit auch gedacht, es wäre durch einen CCU Neustart gelöst gewesen.

Ich habe definitiv nur eine Homeassistant Instanz laufen. Ich habe auch die Logs aus der CCu mal runtergeladen. Auf den ersten Blick sehe ich da nichts besonderes. Hast du ein Stichwort nach dem ich suchen sollte? Derzeitige Logs gehen zurück bis zum 8.11. da war die Meldung auf jeden Fall noch da.

Updates funktionieren eigentlich immer ganz gut, aber manchmal merke ich, dass ein Bewegungsmelder für das Treppenhauslicht nicht reagieren will. Das legt sich dann aber nach einer kurzen Zeit meistens wieder.

Hast du noch eine Idee?

**Reply by SukramJ** [reply]
> Hast du ein Stichwort nach dem ich suchen sollte?

Leider nicht.

> Hast du noch eine Idee?

Nein, aber vielleicht bringt die nächste Version differenzierter Meldungen.

**Comment by tuxianerDE** [comment]
Hey,

hänge mich hier einmal dran. Ich habe ein solches Setup.

Meine CCU (debmatic) ist die Zentrale Instanz für meine Kommunikation mit meinen HomeMatic IP Geräten. Ich habe 2 HA Instanzen.

Die erste HA Instanz verwendet den Instance Name "Debmatic" die zweite Instanz - auf der zweiten HA Instanz, lautet "MieterDebmatic".

Ich bekomme allerdings ebenfalls in der zweiten Instanz konstant die PING/PONG Fehlermeldung und Updates/Events zur Temperatur an Thermostaten bspw. erscheinen nicht im HA. 

Schaue ich auf den falschen Instanznamen oder wo muss ich hinschauen?

Lg
Tuxianer

**Reply by SukramJ** [reply]
Um sicher zugehen gehst Du auf die Integration in der Übersicht und lädtst dir die Disagnoseinformationen herunter. Dort kann man den `instance name` als solchen sehen.

**Reply by tuxianerDE** [reply]
Danke für die schnelle Antwort.

HA Instance 1 -> "instance_name": "Debmatic",
HA Instance 2 -> "instance_name": "MieterDebmatic",

Beide greifen auf die gleiche Debmatic Instanz zu.

Ok die sind also defintiv unterschiedlich. Ich erhalte trotzdem die PING/PONG Message und sehe dass in der zweiten Instanz (MieterDEbmatic) die Updates nicht kommen auch wenn es man es aktiv triggered (Änderung der Temperatur z.B.)

Ich nutze unterschiedliche Nutzer beide mit Admin Rechten.

**Reply by SukramJ** [reply]
Welche Versionen (HA, Integration, CCU) hast Du installiert?
Hast Du die CCU mal neugestartet?
Du solltest vor allem auf der zweiten Instanz in Logs schauen, ob da Fehler zu sehen sind.
Wenn nicht, dann dort mal den DEBUG log in der `configuration.yaml` aktivieren:

```
logger:
  logs:
    hahomematic: info
    hahomematic.caches: debug
```

**Reply by tuxianerDE** [reply]
HA Core: 2023.12.1
Integration: 1.50.0
CCU: Debmatic Firmware: 3.73.9.112

CCU wurde heute morgen neugestartet (update auf die aktuelle Debmatic Version). Ja es gibt Error Logs:
 Pending PONG mismatch: There is a mismatch between send ping events and received pong events for HA instance MieterDebmatic-HmIP-RF. Possible reason 1: You are running multiple instances of HA with the same instance name configured for this integration. Re-add one instance! Otherwise this HA instance will not receive update events from your CCU. Possible reason 2: Something is stuck on the CCU or hasn't been cleaned up. Therefore, try a CCU restart.#033[0m
***

**Reply by tuxianerDE** [reply]
Im Debug, kommt das gleiche:
PING PONG CACHE: Increase pending PING count: MieterDebmatic-BidCos-RF - 16 for ts: 2023-12-13 17:39:16.676841#033[0m

2023-12-13T17:44:47.146452+01:00 troja homeassistantMieter[34130]: #033[36m2023-12-13 17:44:47.143 DEBUG (MainThread) [hahomematic.caches.dynamic] PING PONG CACHE: Removing expired pending PONG: MieterDebmatic-HmIP-RF - 21 for ts: 2023-12-13 17:39:31.687118#033[0m
2023-12-13T17:44:47.147451+01:00 troja homeassistantMieter[34130]: #033[36m2023-12-13 17:44:47.145 DEBUG (MainThread) [hahomematic.caches.dynamic] PING PONG CACHE: Increase pending PING count: MieterDebmatic-HmIP-RF - 21 for ts: 2023-12-13 17:44:47.143316#033[0m


Sehe wie er im Cache hochzählt,a ber wenn die Warning kommt, kommen keine weiteren Infos.

**Reply by SukramJ** [reply]
Hat die zweite Instanz überhaupt schon mal funktioniert?
Ändere nochmal den Logger:

```
logger:
  logs:
    hahomematic: info
    hahomematic.central: debug
```

Kommen da überhaupt Events an?

**Reply by SukramJ** [reply]
Wie läst Du HA laufen? HAOS?

**Reply by tuxianerDE** [reply]
tja so wie es aussieht wohl nie richtig (setup ist aber acuh sehr frisch). 

Beide Instanzen laufen als docker container, dort ist aber acuh das Problem.

network_mode: host

musst gesetzt sein sonst geht es nicht. habe es gerade geändert und jetzt es sauber durch.
